### PR TITLE
First draft

### PR DESCRIPTION
**Pull request by antoine-de**
it seems to work

some cleanup and some work on the parameters need to be done

need #1

### CODE REVIEW

**Comment by antoine-de** [comment]
If nobody is interested I will auto merge the PR :wink:

File: readme.md [hunk 1]
```diff
@@ -1 +1,14 @@
-generic git release script
+# generic git release script
+ 
+ The script perform a git release of a project
```
Comment (pbougue) [inline readme.md]: perform's'

File: release.py [hunk 2]
```diff
@@ -1,35 +1,327 @@
...
+
+        logging.debug("current branch {}".format(self.repo.active_branch))
+
+    def _make_git_release(self, version, prs):
+        """
+        git branch update
+        - a temporary branch is created from the base branch
+        - if some files have been added, they are commited
+        """
+        tmp_name = "release_{version}_{rand}".format(version=version, rand=uuid.uuid4())
```
Comment (kinnou02) [inline release.py:187]: This branch will have to be deleted after the merge to release

File: release.py [hunk 3]
```diff
@@ -1,35 +1,327 @@
...
+        self.generate_debian_changelog = True
         self.release_type = release_type
-        self.repo = git.Repo(".")
+        self.project_path = path
+        self.repo = git.Repo(path)
         self.git = self.repo.git
+        self.remote_name = remote_name
+        self.base_branch = base_branch
+        if github_user and github_token:
+            # TODO auth
```
Comment (datanel) [inline release.py]: you can remove this comment

File: release.py [hunk 4]
```diff
@@ -1,35 +1,327 @@
...
+        logging.info("  git checkout {}; git submodule update".format(self.base_branch))
+        logging.info("  git merge {}".format(RELEASE_BRANCH))
+        logging.info("And when you're happy do:")
+        logging.info("  git push {} {} {} --tags".format(self.remote_name, self.base_branch, RELEASE_BRANCH))
+        #TODO: when we'll be confident, we will do that automaticaly
+
+    def _update_repository(self):
+        """we fetch all latest changes"""
+        logging.info('fetching changes')
+        self.repo.remote(self.remote_name).fetch("--tags")
```
Comment (kinnou02) [inline release.py:247]: you should run a `submodule update` too

File: release.py [hunk 5]
```diff
@@ -1,35 +1,327 @@
...
+    def fetch_labels(self, auth):
+        """ call github to fetch the labels of the pr """
+        if not self._labels:
+            label_query = self.raw_response['_links']['issue']['href'] + '/labels'
+            self._labels = [r['name'] for r in requests.get(label_query, auth=auth).json()]
+        return self._labels
+
+
+def write_debian_changelog(write_lines, project_path):
+    changelog_path = "debian/changelog"
```
Comment (kinnou02) [inline release.py]: It would be nice to use dch

File: release.py [hunk 6]
```diff
@@ -1,35 +1,327 @@
...
+import os
+from shutil import copyfile
+import stat
 from clingon import clingon
+import codecs
+from future.moves import subprocess
 import git
+from git.exc import GitCommandError
+import requests
+import semver
```
Comment (datanel) [inline release.py:13]: it is not in requirements.txt

Comment (antoine-de) [inline release.py:13]: done

File: release.py (hunk 2 above)
Comment (antoine-de) [inline release.py:187]: done

File: release.py (hunk 4 above)
Comment (antoine-de) [inline release.py:247]: I don't think so, I think we need to do it only at the end

File: release.py (hunk 5 above)
Comment (antoine-de) [inline release.py]: you're absolutely right, with it we can be rid of the ugly changelog manipulation.

done,
thanks!

**Pull request merged**